R/summarystatsbayesianztest.R

```diff
@@ -0,0 +1,404 @@
+    summaryTable$addFootnote(.bayesianZTestsGetDefaultBfNote(options))
+  }
+
+
+  summaryTable$addRows(newRow)
+
+  return()
+}
+.bayesianZTestPriorPosteriorPlot <- function(jaspResults, options) {
+
+  if (!is.null(jaspResults[["priorPosteriorPlot"]]))
+    return()
+
+  priorPosteriorPlot <- createJaspPlot(title = gettext("Prior Robustness Plot"), width = 450, height = 300)
+  priorPosteriorPlot$position <- 2
+  priorPosteriorPlot$dependOn(c(.bayesianZTestsDependencies, c("plotPriorAndPosterior", "plotPriorAndPosteriorAdditionalInfo")))
+  jaspResults[["priorPosteriorPlot"]] <- priorPosteriorPlot
+
+  data      <- jaspResults[["data"]][["object"]]
+  alt       <- switch(
+    options[["hypothesis"]],
+    "notEqualToTestValue"  = "two.sided",
+    "greaterThanTestValue" = "greater",
+    "lessThanTestValue"    = "less"
+  )
+
+  BF10 <- .bayesianZTestsComputeBf10(
+    x  = data[["es"]], se = data[["se"]],
+    mu = options[["priorMean"]], sigma = options[["priorSd"]],
+    alternative = alt
+  )
+  plotData <- .bayesianZTestsComputePlot(
+    x  = data[["es"]], se = data[["se"]],
+    mu = options[["priorMean"]], sigma = options[["priorSd"]],
+    alternative = alt,
+    isLog = options[["dataType"]] == "esAndCiLog"
+  )
+
+  plot <- jaspGraphs::PlotPriorAndPosterior(
+    dfLines    = plotData[["dfLines"]],
+    dfPoints   = plotData[["dfPoints"]],
+    BF         = if(options[["plotPriorAndPosteriorAdditionalInfo"]]) {if (options[["bayesFactorType"]] == "BF01") 1/BF10 else BF10},
+    bfType     = if(options[["plotPriorAndPosteriorAdditionalInfo"]]) {if (options[["bayesFactorType"]] == "BF01") "BF01" else "BF10"},
+    hypothesis = switch(
+      options[["hypothesis"]],
+      "notEqualToTestValue"  = "equal",
+      "greaterThanTestValue" = "greater",
+      "lessThanTestValue"    = "smaller"
+    ),
+    xName      = if (options[["dataType"]] == "esAndCiLog") gettext("exp(Effect size)") else gettext("Effect size")
+  )
+
+  priorPosteriorPlot[["plotObject"]] <- plot
+
+  return()
+}
+.bayesianZTestRobustnessPlot     <- function(jaspResults, options) {
+
+  if (!is.null(jaspResults[["robustnessPlot"]]))
+    return()
+
+  robustnessPlot <- createJaspPlot(title = gettext("Bayes Factor Robustness Plot"), width = 550, height = 450)
+  robustnessPlot$position <- 3
+  robustnessPlot$dependOn(c(.bayesianZTestsDependencies, c("plotBayesFactorRobustness" ,"robustnessPriorMeanMin", "robustnessPriorMeanMax", "robustnessPriorSdMin", "robustnessPriorSdMax", "plotBayesFactorRobustnessContours", "plotBayesFactorRobustnessContoursValues")))
+  jaspResults[["robustnessPlot"]] <- robustnessPlot
+
+  # specify contour breaks
+  formatBF <- function(BF) ifelse(BF < 1, paste0("1/", round(1/BF, 2)), as.character(round(BF, 2)))
+  if (options[["plotBayesFactorRobustnessContours"]]) {
+    bfBreaks <- strsplit(options[["plotBayesFactorRobustnessContoursValues"]], ",")[[1]]
+    bfBreaks <- sapply(bfBreaks, function(bfBreak) eval(parse(text = bfBreak)))
```

Comment:
```suggestion
    bfBreaks <- vapply(bfBreaks, FUN = function(bfBreak) eval(parse(text = bfBreak)), FUN.VALUE = numeric(1))
```

Reply:
any specific reason for using vapply there?

Reply:
yes, always use `vapply` (or something from `purrr`) over `sapply` so that if it fails it fails early (defensive programming). But see the bug below, you're parsing user input as code without validation, which is a bit tricky. I wonder if there is a better way to do this.

Reply:
In fact, `sapply` is set to become one of the unwanted functions once we have [lintr](https://github.com/r-lib/lintr) set up.